src/main/java/org/fogbowcloud/saps/engine/core/scheduler/Scheduler.java

```diff
@@ -85,33 +93,52 @@ public void start() {
-                List<JobResponseDTO> jobsWithEqualJobName = getJobByNameInArrebol(jobName,
-                        "gets job by name [" + jobName + "]");
-                if (jobsWithEqualJobName.size() == 0)
+            while(true) {
+                try {
+                    LOGGER.info("Recovering Task [" + task.getTaskId() + "]");
+                    recovery(task);
+                    break;
+                } catch (ConnectException e) {
+                    LOGGER.error(e);
+                    try {
+                        LOGGER.info("Sleeping for " + retryPeriodSec + " seconds");
+                        Thread.sleep(retryPeriodSec * 1000);
+                    } catch (InterruptedException ie) {
+                        LOGGER.error(ie);
+                    }
+                } catch (Exception e) {
+                    LOGGER.error("Error while recovering task [" + task.getTaskId() + "]");
                     rollBackTaskState(task);
-                else if (jobsWithEqualJobName.size() == 1) { // TODO add check jobName ==
-                    // jobsWithEqualJobName.get(0).get...
-                    String arrebolJobId = jobsWithEqualJobName.get(0).getId();
-                    updateStateInCatalog(task, task.getState(), SapsImage.AVAILABLE, SapsImage.NON_EXISTENT_DATA,
-                            arrebolJobId,
-                            "updates task [" + task.getTaskId() + "] with Arrebol job ID [" + arrebolJobId + "]");
-                    tasksForPopulateSubmittedJobList.add(task);
-                } else {
-                    // TODO ????
                 }
-            } else {
-                String arrebolJobId = task.getArrebolJobId();
-                arrebol.addJobInList(new JobSubmitted(arrebolJobId, task));
             }
         }
+    }
 
-        arrebol.populateJobList(tasksForPopulateSubmittedJobList);
+    private void recovery(SapsImage task) throws Exception {
+        if (task.getArrebolJobId().equals(SapsImage.NONE_ARREBOL_JOB_ID)) {
+            String jobName = task.getState().getValue() + "-" + task.getTaskId();
+            List<JobResponseDTO> jobsWithEqualJobName;
+            jobsWithEqualJobName = jobExecutionService.getJobByLabel(jobName);
+            if (jobsWithEqualJobName.size() == 0)
+                rollBackTaskState(task);
+            else if (jobsWithEqualJobName.size() == 1) { // TODO add check jobName ==
+                // jobsWithEqualJobName.get(0).get...
+                String arrebolJobId = jobsWithEqualJobName.get(0).getId();
+                updateStateInCatalog(task, task.getState(), SapsImage.AVAILABLE, SapsImage.NON_EXISTENT_DATA,
+                    arrebolJobId,
+                    "updates task [" + task.getTaskId() + "] with Arrebol job ID [" + arrebolJobId + "]");
+                this.submittedJobsID.add(new JobSubmitted(task.getArrebolJobId(), task));
+            } else {
+                // TODO ????
```

Comment:
Change the message of this TODO to "discuss what action to take in this situation. Should I remove all jobs found from the execution service and do a rollback in task? Or should I take advantage of any of them?"

Reply:
Done.